### DISCUSSION TITLE
TCases vs ReadyAPI

### DISCUSSION THREAD

**Discussion by AnirudhChaudhary** [Cornutum/tcases#205] [Q&A] [answered]
Hey Kerry,

I was looking at a different tool called ReadyAPI by SmartBear that also seems to automate REST API test creation. I was wondering whether you were aware of this tool and had any insights about the differences between your tool and theirs. Just curious.

Thanks.

**Comment by kerrykimbrough** [comment, marked as answer]
I've never used ReadyAPI. But after looking over the ReadyAPI web site, here is how I would compare it to Tcases for OpenAPI.
  * RA is big. TFOA is small. RA has lots of features for building API tests. TFOA just does one thing: generate functional tests for API requests.
  * RA offers lots of ways to build tests. TFOA offers just one: generate them from an OpenAPI definition
  * Converts an OpenAPI definition into executable tests? TFOA: yes, RA: ? I couldn't find any details on if or how RA might do this.
  * RA measures coverage. TFOA guarantees coverage. Tests generated by TFOA automatically provide a strong type of coverage: every request is executed with every significant variation of values for every request input. RA will report which requests and parameters have not been used, but it's up to you to add coverage to your tests. Also, the type of coverage measured by RA seems to be weaker than the type provided by TFOA.
